src/devices/cpu/mcs51/mcs51.cpp

```diff
@@ -2469,6 +2462,51 @@ uint8_t ds5002fp_device::sfr_read(size_t offset)
 	return m_data->read_byte((size_t) offset | 0x100);
 }
 
+/*
+Documentation states that having the battery connected "maintains the internal scratchpad RAM" and "certain SFRs"
+(although it isn't clear exactly which SFRs except for those explicitly mentioned)
+*/
+
+void ds5002fp_device::nvram_default()
+{
+	memset( m_scratchpad, 0, 0x80 );
+	memset( m_sfr_ram, 0, 0x80 );
+
+	int expected_bytes = 0x80 + 0x80;
+
+	if (!m_region.found())
+	{
+		logerror( "ds5002fp_device(%s) region not found\n", tag() );
```

Comment:
this is straight from the other nvram devices, you should probably clean them up if you don't want people copying them.